Appz/Appz/Apps/Mail.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Mail.swift
+//  Appz
+//
+//  Created by Suraj Shirvankar on 12/5/15.
+//  Copyright © 2015 kitz. All rights reserved.
+//
+
+import Foundation
+
+
+public extension AvailableApplications {
+    
+    public struct Mail: ExternalApplication {
+        
+        public let scheme = "mailto:"
+        public let fallbackURL: String? = ""
+    }
+    
+    public func mail(action: Mail.Action) -> Bool {
+        return appCaller.launch(Mail(), action: action)
+    }
+}
+
+// MARK: - Actions
+
+public extension AvailableApplications.Mail {
+    
+    public enum Action {
+        case Email(email: String, subject: String, body: String)
```

Comment:
You make a good point, yet approach is always to try and aim for something "scalable", design-wise. Writing definitions for every possible permutation goes against that.

I thought about this, and I think the action can be `Compose`, since you are composing an email. Then, it takes an `Email` struct, which is a simple struct that we can then use default arguments with:

```
app.open.mail(.Compose(Email(subject: "Reads like english, doesn't it?")))
``